src/components/common/YourCompany/StartDate.vue

```diff
@@ -0,0 +1,231 @@
+              <v-btn text small color="primary" id="start-menu-btn" v-on="on">
+                <v-icon id="start-menu">{{dropdown ? 'mdi-menu-up' : 'mdi-menu-down'}}</v-icon>
+              </v-btn>
+            </template>
+            <v-btn text color="primary" id="more-changes-btn"
+              @click="onChangeClicked(); dropdown = false">
+              <v-icon small color="primary">mdi-pencil</v-icon>
+              <span>{{ editLabel }}</span>
+            </v-btn>
+          </v-menu>
+        </span>
+
+      </v-col>
+    </v-row>
+  </section>
+</template>
+
+<script lang="ts">
+import { Component, Mixins, Prop, Watch } from 'vue-property-decorator'
+import { Getter, Action } from 'vuex-class'
+import { CommonMixin, DateMixin } from '@/mixins/'
+import { DatePicker as DatePickerShared } from '@bcrs-shared-components/date-picker'
+import { ActionBindingIF } from '@/interfaces'
+
+@Component({
+  components: {
+    DatePickerShared
+  }
+})
+export default class StartDate extends Mixins(CommonMixin, DateMixin) {
+  // Global getters
+  @Getter isBenIaCorrectionFiling!: boolean
+  @Getter isFirmCorrectionFiling!: boolean
+  @Getter getCorrectedStartDate!: string
+  @Getter getBusinessFoundingDate!: string // actually date-time
+  @Getter getCurrentDate!: string
+  @Getter getCurrentJsDate!: Date
+
+  // Global setter
+  @Action setCorrectionStartDate!: ActionBindingIF
+  @Action setValidComponent!: ActionBindingIF
+
+  /** Whether to show invalid section styling. */
+  @Prop({ default: false })
+  readonly invalidSection: boolean
+
+  protected dropdown = false
+  protected onEditMode = false
+  protected isCorrected = !!this.getCorrectedStartDate
+  protected newCorrectedStartDate = null as string // date is "Month Day, Year"
+
+  /** The minimum start date that can be entered (Up to 2 years ago today). */
+  protected get startDateMin (): Date {
+    const startDateMin = new Date(this.getCurrentJsDate)
+    startDateMin.setFullYear(startDateMin.getUTCFullYear() - 2)
+    startDateMin.setHours(0, 0, 0) // Set time to 0 for accurate Date Rules comparison
+
+    return startDateMin
+  }
+
+  /** The minimum start date string. */
+  protected get startDateMinStr (): string {
+    return this.dateToYyyyMmDd(this.startDateMin)
+  }
+
+  /** The maximum start date that can be entered (Up to 90 days from today). */
+  protected get startDateMax (): Date {
+    const startDateMax = new Date(this.getCurrentJsDate)
+    startDateMax.setDate(startDateMax.getUTCDate() + 90)
+    return startDateMax
+  }
+
+  /** The maximum start date string. */
+  protected get startDateMaxStr (): string {
+    return this.dateToYyyyMmDd(this.startDateMax)
+  }
+
+  /** The business date or business start date string. */
+  protected get businessStartDate (): string {
+    if (
+      this.isFirmCorrectionFiling ||
+      this.isFirmChangeFiling ||
+      this.isFirmConversionFiling
+    ) {
+      if (this.getCorrectedStartDate) {
```

Comment:
Would it make sense to use hasBusinessStartDateChanged here?

Reply:
Yes I changed it.